code/modules/mob/living/emote.dm

```diff
@@ -287,14 +294,18 @@
 	return ..() && user.can_speak(allow_mimes = TRUE)
 
 /datum/emote/living/laugh/get_sound(mob/living/user)
-	if(!ishuman(user))
+	if(!ishuman(user) || !islizard(user) || !ismoth(user))
 		return
 	var/mob/living/carbon/human/human_user = user
 	if((ishumanbasic(human_user) || isfelinid(human_user)) && !HAS_MIND_TRAIT(human_user, TRAIT_MIMING))
 		if(human_user.gender == FEMALE)
 			return 'sound/voice/human/womanlaugh.ogg'
 		else
 			return pick('sound/voice/human/manlaugh1.ogg', 'sound/voice/human/manlaugh2.ogg')
+	if(islizard(human_user) && !HAS_MIND_TRAIT(human_user, TRAIT_MIMING))
+		return 'sound/voice/lizard/lizard_laugh1.ogg'
+	if(ismoth(human_user) && !HAS_MIND_TRAIT(human_user, TRAIT_MIMING))
+		return 'sound/voice/moth/moth_laugh1.ogg'
```

Comment:
This sound should be on the species datum or (even better) on the tongue organ.

Reply:
I have no idea how to do this, can you give some directions?

Reply:
Add a new proc either to the species datum or the tongue which returns a sound file (or nothing).
For humans and felinids (or their tongues) have it return 
```
		if(human_user.gender == FEMALE)
			return 'sound/voice/human/womanlaugh.ogg'
		else
			return pick('sound/voice/human/manlaugh1.ogg', 'sound/voice/human/manlaugh2.ogg')
```
For lizards have it 
````
return 'sound/voice/lizard/lizard_laugh1.ogg'
```
and so on.

In _this_ proc, retrieve either the species (from the DNA of a human) or the tongue (by checking the organs of a human) and call the proc to get the appropriate sound file.

Look at the scream emote for an example of how it is done using the species datum.